### PR TITLE
[build-utils] Optimize lockfile version detection by reading only first 10 lines

### PR DESCRIPTION
**Pull request by ofhouse**
## Summary

Optimizes lockfile version detection by reading only the first 10 lines instead of parsing entire lockfiles. This improves performance for large `pnpm-lock.yaml` and `package-lock.json` files.

## Changes

- Added `readLockfileVersion()` function that reads only the first 10 lines of lockfiles
- Uses filename-based pattern matching to determine the extraction strategy:
  - `pnpm-lock.yaml`: Extracts `lockfileVersion: 'X.X'` from line 1
  - `package-lock.json`: Extracts `"lockfileVersion": X` from header
- Falls back to full file parsing for unknown lockfile types or edge cases
- Updated `scanParentDirs()` to use the new optimized function

## Testing

- Added 10 unit tests covering pnpm-lock.yaml, package-lock.json, non-existent files, and fallback behavior
- All existing unit tests pass

### CODE REVIEW

File: packages/build-utils/src/fs/read-config-file.ts [hunk 1]
```diff
@@ -61,3 +62,99 @@ export async function getPackageJson(dir: string): Promise<PackageJson> {
...
+  }
+
+  // If we found the version in the header, return it
+  if (lockfileVersion !== null) {
+    return { lockfileVersion };
+  }
+
+  // Fall back to full file parsing for unknown formats or if not found in header
+  return readConfigFile<{ lockfileVersion: number }>(filePath);
+}
```
[the remark this answers is not in the document]
Comment (ofhouse) [reply packages/build-utils/src/fs/read-config-file.ts:170]: Fixed in https://github.com/vercel/vercel/pull/14657/changes/6ec2064dbc2a8ff0808b8e6c31251e0cafc81b4e
